Lombiq.Tests.UI/Services/UITestExecutor.cs

```diff
@@ -51,12 +55,25 @@ private static async Task ExecuteOrchardCoreTestInnerAsync(UITestManifest testMa
 
         configuration.TestOutputHelper.WriteLineTimestampedAndDebug("Finished preparation for {0}.", testManifest.Name);
 
+        if (_semaphoreSlim == null && configuration.MaxRunningConcurrentTests > 0)
+        {
+            lock (_initLock)
+            {
+                _semaphoreSlim ??= new SemaphoreSlim(configuration.MaxRunningConcurrentTests);
+            }
+        }
+
         var retryCount = 0;
         var passed = false;
         while (!passed)
         {
             try
             {
+                if (_semaphoreSlim != null)
```

Comment:
This can never be true here though, right? On line 92 too.

Reply:
_semaphoreslim will be set on line 62. So it won't be null, which means there will be a limit and `await _semaphoreSlim.WaitAsync();` will be run. 

It will be null only when you want to set it to unlimited number of concurrent tests with passing `-1` to `MaxRunningConcurrentTests`.

Reply:
Right.